src/Mono.Android/Android.Security/KeyChain.cs

```diff
@@ -0,0 +1,94 @@
+#nullable enable
+using System;
+using System.Runtime.Versioning;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace Android.Security
+{
+	public partial class KeyChain
+	{
+		[SupportedOSPlatform("android14.0")]
+		public static X509Certificate2? GetCertificateWithPrivateKey (Android.Content.Context context, string alias)
+		{
+			var privateKey = KeyChain.GetPrivateKey (context, alias);
+			if (privateKey is null) {
+				return null;
+			}
+
+			var chain = KeyChain.GetCertificateChain (context, alias);
+			if (chain is null) {
+				return null;
+			}
+
+			var privateKeyEntry = new Java.Security.KeyStore.PrivateKeyEntry (privateKey, chain);
+			return new X509Certificate2 (privateKeyEntry.Handle);
```

Comment:
As-is, this is begging the GC to make life painful for you.  A possible serialization of execution is:

 1. `var privateKeyEntry = new Java.Security.KeyStore.PrivateKeyEntry (privateKey, chain);`
 2. `var __tmp = privateKeyEntry.Handle;`
 3. GC determines that `privateKeyEntry` is no longer referenced, finalizes the instance, `__tmp` is invalidated.
 4. `new X509Certificate2(__tmp)` now attempts to use an invalid GREF.  BOOM goes the app.

Is this unlikely?  Yes.  Can it happen?  Oh yes: see also e88cfbcf9ff4f4e179a60f1e8af2bef3537a66b4, which took *ages* to figure out what was going wrong…

Fortunately the fix is straightforward: add a `GC.KeepAlive(privateKeyEntry)`:

```csharp
var privateKeyEntry = new Java.Security.KeyStore.PrivateKeyEntry (privateKey, chain);
var cert = new X509Certificate2 (privateKeyEntry.Handle);
GC.KeepAlive (privateKeyEntry);
return cert;
```